TransformationSystem/Utilities/ReplicationTransformation.py

```diff
@@ -42,6 +42,8 @@ def createDataTransformation(flavour, targetSE, sourceSE,
   if isinstance(targetSE, six.string_types):
     targetSE = [targetSE]
 
+  sourceSE = '%s' % (",".join(sourceSE))
```

Comment:
It doesn't look like a good idea:
```
In [1]: sourceSE = 'someSE'

In [2]: ",".join(sourceSE)
Out[2]: 's,o,m,e,S,E'
```

Also:
- why do you want to have multiple source SEs?
- I think the lines just above about `targetSE` should be:
```
  if isinstance(targetSE, six.string_types):
    targetSE = targetSE.split(',')
```

Reply:
> It doesn't look like a good idea:
> 
> ```
> In [1]: sourceSE = 'someSE'
> 
> In [2]: ",".join(sourceSE)
> Out[2]: 's,o,m,e,S,E'
> ```
>

I don't know if there are other use cases, but I've found that when script uses this utility is used by the  dirac-transformation-replication script, sourceSE is of type 'list' and without this fix the corresponding Requests are not created:

```
2020-12-07 13:58:32 UTC Transformation/RequestTaskAgent/ [Thread14]
[2316] .execute ERROR: Failed to submit task to RMS
ReqClient.putRequest: unable to set request '00002316_00000030'
2020-12-07 13:58:38 UTC
Transformation/RequestTaskAgent/RequestManagement/ReqClient/pid_29415
WARN: putRequest: unable to set request '00002316_00000031' at
RequestManager putRequest: unexpected exception
(_mysql_exceptions.OperationalError) (1241, 'Operand should contain 1
column(s)')
[SQL: INSERT INTO `Operation` (`TargetSE`, `CreationTime`,
`SourceSE`, `Arguments`, `Error`, `Type`, `Order`, `Status`,
`LastUpdate`, `SubmitTime`, `Catalog`, `RequestID`) VALUES (%s, %s,
%s, %s, %s, %s, %s, %s, %s, %s, %s, %s)]
[parameters: ([u'CC-IN2P3-Tape'], datetime.datetime(2020, 12, 7, 13,
54, 8), [u'CC-IN2P3-Disk', u'DESY-ZN-Disk', u'CYF-STORM-Disk'], None,
None, 'ReplicateAndRegister', 0, 'Waiting', datetime.datetime(2020,
12, 7, 13, 54, 8), datetime.datetime(2020, 12, 7, 13, 54, 8), None,
1351909L)]
(Background on this error at:http://sqlalche.me/e/13/e3q8)
2020-12-07 13:58:38 UTC
Transformation/RequestTaskAgent/RequestManagement/ReqClient/pid_29415
ERROR: putRequest: unable to set request '00002316_00000031'
```


> Also:
> 
>     * why do you want to have multiple source SEs?
> 

This is already foreseen by the dirac-transformation-replication script:

....

```

Options:
  -G  --GroupSize <value>      : Number of Files per transformation task
  -R  --GroupName <value>      : TransformationGroup Name
  -S  --SourceSEs <value>      : SourceSE(s) to use, comma separated list
  -N  --Extraname <value>      : String to append to transformation name
  -P  --Plugin <value>         : Plugin to use for transformation
  -T  --Flavour <value>        : Flavour to create: Replication or Moving
  -K  --MetaKey <value>        : Meta Key to use: TransformationID
  -M  --MetaData <value>       : MetaData to use Key/Value Pairs: 'DataType:REC,'
  -x  --Enable                 : Enable the transformation creation, otherwise dry-run
```

Our use case for instance is that we want to move data from a sourceSE to a targetSE, but we only know the list of sourceSE where the file is potentially hosted. 
That means it could be located in SE1, SE2 or SE3 and we want to move it to SE4, so we create a transformation from SE1,SE2,SE3 to SE4.
We have verified in production that with this fix the 'Moving' transformations work fine (and the same would be for 'Replication' transformations).


>     * I think the lines just above about `targetSE` should be:
> 
> 
> ```
>   if isinstance(targetSE, six.string_types):
>     targetSE = targetSE.split(',')
> ```

ok. I haven't tried with multiple targetSE, so that's probably why I have not observed any issue.

Reply:
Fine, I think you can simply protect the code with the usual 
```
if isinstance(sourceSE, six.string_types):
  something
elif isinstance(sourceSE, (list, tuple)):
  somethingElse
```
and then update the docstring of the method.

Reply:
OK, I think it's enough to protect in this way:

```
if isinstance(sourceSE, (list, tuple)):
  sourceSE = '%s' % (",".join(sourceSE))
```
Do you agree?

Reply:
Should be.